File: poll_agent.js
const http = require('http');

const SESSION_ID = 'session-1773060372155.md';
const SERVER_URL = 'http://localhost:7439';
const POLL_INTERVAL = 30000; // 30 seconds
const MAX_ITERATIONS = 100;

console.log('Starting Alice Agent Poll Loop');
console.log(`SessionID: ${SESSION_ID}`);
console.log(`Server: ${SERVER_URL}`);
console.log(`Poll Interval: ${POLL_INTERVAL}ms`);
console.log('---\n');

// First, check if server is running
console.log('Step 1: Checking if server is running...');
const checkReq = http.get(`${SERVER_URL}/agent/input?sessionId=${SESSION_ID}`, (res) => {
  let data = '';
  res.on('data', d => data += d);
  res.on('end', () => {
    console.log(`✓ Server responded with status ${res.statusCode}`);
    try {
      const json = JSON.parse(data);
      console.log('Initial Response:', JSON.stringify(json, null, 2));
    } catch(e) {
      console.log('Response:', data);
    }
    console.log('\nStep 2: Starting POLL loop...\n');
    startPolling();
  });
});

checkReq.on('error', (e) => {
  console.log(`✗ Server not reachable: ${e.message}`);
  process.exit(1);
});

checkReq.end();

function startPolling() {
  let count = 0;
  
  function poll() {
    if (count >= MAX_ITERATIONS) { 
      console.log(`\n✗ POLL TIMEOUT - reached ${MAX_ITERATIONS} iterations (${MAX_ITERATIONS * 30 / 60} minutes)`);
      process.exit(0);
      return; 
    }
    
    count++;
    const timestamp = new Date().toISOString();
    console.log(`[${timestamp}] Iteration ${count}/${MAX_ITERATIONS}: Polling for new tasks...`);
    
    const getReq = http.get(`${SERVER_URL}/agent/input?sessionId=${SESSION_ID}`, (res) => {
      let data = '';
      res.on('data', d => data += d);
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          console.log(`  Response: ${JSON.stringify(json)}`);
          
          if (json.hasContent && json.content) { 
            console.log(`\n✓ NEW_TASK RECEIVED:`);
            console.log(`  Content: ${json.content}`);
            console.log(`  Full Response: ${JSON.stringify(json, null, 2)}`);
            process.exit(0);
            return;
          }
        } catch(e) {
          console.log(`  Parse Error: ${e.message}`);
        }
        
        // Schedule next poll
        const nextTime = new Date(Date.now() + POLL_INTERVAL).toISOString();
        console.log(`  Waiting until ${nextTime}...\n`);
        setTimeout(poll, POLL_INTERVAL);
      });
    });
    
    getReq.on('error', (e) => { 
      console.log(`  Error: ${e.message}`);
      console.log(`  Retrying in ${POLL_INTERVAL/1000}s...\n`);
      setTimeout(poll, POLL_INTERVAL);
    });
    
    getReq.end();
  }
  
  poll();
}

File: poll_agent.py
#!/usr/bin/env python3
import http.client
import json
import time
from datetime import datetime

SESSION_ID = 'session-1773060372155.md'
SERVER_HOST = 'localhost'
SERVER_PORT = 7439
POLL_INTERVAL = 30  # seconds
MAX_ITERATIONS = 100

print('Starting Alice Agent Poll Loop')
print(f'SessionID: {SESSION_ID}')
print(f'Server: http://{SERVER_HOST}:{SERVER_PORT}')
print(f'Poll Interval: {POLL_INTERVAL}s')
print('---\n')

# First, check if server is running
print('Step 1: Checking if server is running...')
try:
    conn = http.client.HTTPConnection(SERVER_HOST, SERVER_PORT, timeout=10)
    conn.request('GET', f'/agent/input?sessionId={SESSION_ID}')
    response = conn.getresponse()
    data = response.read().decode('utf-8')
    print(f'✓ Server responded with status {response.status}')
    try:
        json_data = json.loads(data)
        print('Initial Response:', json.dumps(json_data, indent=2))
    except:
        print('Response:', data)
    conn.close()
except Exception as e:
    print(f'✗ Server not reachable: {e}')
    exit(1)

print('\nStep 2: Starting POLL loop...\n')

count = 0
while count < MAX_ITERATIONS:
    count += 1
    timestamp = datetime.now().isoformat()
    print(f'[{timestamp}] Iteration {count}/{MAX_ITERATIONS}: Polling for new tasks...')
    
    try:
        conn = http.client.HTTPConnection(SERVER_HOST, SERVER_PORT, timeout=10)
        conn.request('GET', f'/agent/input?sessionId={SESSION_ID}')
        response = conn.getresponse()
        data = response.read().decode('utf-8')
        
        try:
            json_data = json.loads(data)
            print(f'  Response: {json.dumps(json_data)}')
            
            if json_data.get('hasContent') and json_data.get('content'):
                print(f'\n✓ NEW_TASK RECEIVED:')
                print(f'  Content: {json_data.get("content")}')
                print(f'  Full Response: {json.dumps(json_data, indent=2)}')
                conn.close()
                break
        except json.JSONDecodeError as e:
            print(f'  Parse Error: {e}')
            print(f'  Raw Response: {data}')
        
        conn.close()
        
        # Schedule next poll
        next_time = datetime.fromtimestamp(time.time() + POLL_INTERVAL).isoformat()
        print(f'  Waiting until {next_time}...\n')
        time.sleep(POLL_INTERVAL)
        
    except Exception as e:
        print(f'  Error: {e}')
        print(f'  Retrying in {POLL_INTERVAL}s...\n')
        time.sleep(POLL_INTERVAL)

if count >= MAX_ITERATIONS:
    print(f'\n✗ POLL TIMEOUT - reached {MAX_ITERATIONS} iterations ({MAX_ITERATIONS * POLL_INTERVAL / 60:.0f} minutes)')
